camayoc/tests/remote/rho/test_scan.py

```diff
@@ -13,35 +13,236 @@
 :testtype: functional
 :upstream: yes
 """
+import csv
+import itertools
+import os
+import time
+from io import BytesIO
 
+import pexpect
 import pytest
 
-from camayoc import config
+from camayoc import cli, utils
+from camayoc.config import get_config
+from camayoc.constants import (
+    CONNECTION_PASSWORD_INPUT,
+    RHO_ALL_FACTS,
+    RHO_PRIVILEGED_FACTS,
+    VCENTER_CLUSTER,
+    VCENTER_DATA_CENTER,
+    VCENTER_HOST,
+)
 from camayoc.exceptions import ConfigFileNotFoundError
-from camayoc.constants import RHO_PRIVILEGED_FACTS
+from camayoc.tests.rho.utils import auth_add, input_vault_password
 
 
-def profiles():
-    """Gather profiles from config file."""
+SCAN_RESULTS = {}
+"""Cache for the scan results returned by :func:get_scan_result."""
+
+
+def is_live(client, server, num_pings=10):
+    """Test if server responds to ping.
+
+    Returns true if server is reachable, false otherwise.
+    """
+    client.response_handler = cli.echo_handler
+
+    ping = client.run(('ping', '-c', num_pings, server))
+    return ping.returncode == 0
+
+
+def wait_until_live(servers, timeout=60):
+    """Wait for servers to be live.
+
+    For each server in the "servers" list, verify if it is reachable.
+    Keep trying until a connection is made for all servers or the timeout
+    limit is reached.
+
+    If the timeout limit is reached, we exit even if there are unreached hosts.
+    This means tests could fail with "No auths valid for this profile" if every
+    host in the profile is unreachable. Otherwise, if there is at least one
+    valid host, the scan will go on and only facts about reached hosts will be
+    tested.
+
+    `See rho issue #302 <https://github.com/quipucords/rho/issues/302>`_
+    """
+    system = cli.System(hostname='localhost', transport='local')
+    client = cli.Client(system)
+
+    unreached = servers
+    while unreached and timeout > 0:
+        unreached = [host for host in unreached if not is_live(client, host)]
+        time.sleep(10)
+        timeout -= 10
+
+
+@pytest.fixture(scope='module')
+def vcenter_client():
+    """Create a vCetner client."""
```

Comment:
All new functions docstring needs to be properly defined, will work on that after having all the needed code in place.